features/roomdetails/impl/src/test/kotlin/io/element/android/features/roomdetails/members/RoomMemberListPresenterTests.kt

```diff
@@ -20,62 +20,132 @@ import app.cash.molecule.RecompositionClock
 import app.cash.turbine.test
 import com.google.common.truth.Truth
+import io.element.android.features.roomdetails.aMatrixRoom
+import io.element.android.features.roomdetails.impl.members.RoomMemberListDataSource
+import io.element.android.features.roomdetails.impl.members.RoomMemberListEvents
 import io.element.android.features.roomdetails.impl.members.RoomMemberListPresenter
-import io.element.android.features.userlist.api.SelectionMode
-import io.element.android.features.userlist.api.UserListDataSource
-import io.element.android.features.userlist.api.UserListDataStore
-import io.element.android.features.userlist.api.UserListPresenter
-import io.element.android.features.userlist.api.UserListPresenterArgs
-import io.element.android.features.userlist.api.UserSearchResultState
-import io.element.android.features.userlist.impl.DefaultUserListPresenter
-import io.element.android.features.userlist.test.FakeUserListDataSource
+import io.element.android.features.roomdetails.impl.members.RoomMemberSearchResultState
+import io.element.android.features.roomdetails.impl.members.RoomMembers
+import io.element.android.features.roomdetails.impl.members.aRoomMember
+import io.element.android.features.roomdetails.impl.members.aRoomMemberList
+import io.element.android.features.roomdetails.impl.members.aVictor
+import io.element.android.features.roomdetails.impl.members.aWalter
 import io.element.android.libraries.architecture.Async
-import io.element.android.libraries.matrix.test.room.FakeMatrixRoom
-import io.element.android.libraries.matrix.ui.components.aMatrixUser
+import io.element.android.libraries.matrix.api.core.UserId
+import io.element.android.libraries.matrix.api.room.MatrixRoomMembersState
+import io.element.android.libraries.matrix.api.room.RoomMembershipState
 import io.element.android.tests.testutils.testCoroutineDispatchers
+import kotlinx.collections.immutable.toImmutableList
 import kotlinx.coroutines.ExperimentalCoroutinesApi
 import kotlinx.coroutines.test.runTest
-import okhttp3.internal.toImmutableList
 import org.junit.Test
 
 @ExperimentalCoroutinesApi
 class RoomMemberListPresenterTests {
 
-    private val testCoroutineDispatchers = testCoroutineDispatchers()
-
     @Test
-    fun `present - search is done automatically on start, but is async`() = runTest {
-        val searchResult = listOf(aMatrixUser())
-        val userListDataSource = FakeUserListDataSource().apply {
-            givenSearchResult(searchResult)
+    fun `search is done automatically on start, but is async`() = runTest {
+        val presenter = RoomMemberListPresenter(
+            roomMemberListDataSource = RoomMemberListDataSource(
+                room = aMatrixRoom().apply {
+                    givenRoomMembersState(MatrixRoomMembersState.Ready(aRoomMemberList()))
+                },
+                coroutineDispatchers = testCoroutineDispatchers()
+            ),
+            coroutineDispatchers = testCoroutineDispatchers()
+        )
```

Comment:
This is a bit hard to read and it's repeated in every test case. Maybe we could centralize this instantiation in a couple of functions? Something like:

```kotlin
private fun createDataSource(
    room: Room = aMatrixRoom(), 
    coroutineDispatchers: Dispatchers = testCoroutineDispatchers()
) = RoomMemberListDataSource(room, coroutineDispatchers)

private fun createPresenter(
    dataSource:  RoomMemberListDataSource = createDataSource(),
    coroutineDispatchers: Dispatchers = testCoroutineDispatchers()
) = RoomMemberListPresenter(dataSource, coroutineDispatchers)
```

And you can provide alternatives to the default parameters if needed.